skyportal/handlers/api/mmadetector.py

```diff
@@ -610,3 +613,396 @@ def delete(self, spectrum_id):
+                    MMADetectorTimeInterval.end_time <= observed_before
+                )
+
+            if observed_after:
+                time_interval_query = time_interval_query.where(
+                    MMADetectorTimeInterval.start_time >= observed_after
+                )
+
+            time_intervals = session.scalars(time_interval_query).unique().all()
+            data = []
+            for time_interval in time_intervals:
+                seg = time_interval.time_interval
+                data.append(
+                    {
+                        'id': time_interval.id,
+                        'time_interval': [seg.lower, seg.upper],
+                        'owner': time_interval.owner,
+                        'groups': time_interval.groups,
+                        'detector': time_interval.detector,
+                    }
+                )
+            return self.success(data=data)
+
+    @permissions(['Upload data'])
+    def patch(self, time_interval_id):
+        """
+        ---
+        description: Update mmadetector time_interval
+        tags:
+          - mmadetector_time_intervals
+        parameters:
+          - in: path
+            name: time_interval_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema: MMADetectorTimeIntervalNoID
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        try:
+            time_interval_id = int(time_interval_id)
+        except TypeError:
+            return self.error('Could not convert time_interval id to int.')
+
+        data = self.get_json()
+        group_ids = data.pop("group_ids", None)
+        if group_ids == 'all':
+            group_ids = [g.id for g in self.current_user.accessible_groups]
+
+        with self.Session() as session:
+            stmt = MMADetectorTimeInterval.select(self.current_user).where(
+                MMADetectorTimeInterval.id == time_interval_id
+            )
+            time_interval = session.scalars(stmt).first()
+
+            if group_ids:
+                groups = (
+                    session.scalars(
+                        Group.select(self.current_user).where(Group.id.in_(group_ids))
+                    )
+                    .unique()
+                    .all()
+                )
+                if {g.id for g in groups} != set(group_ids):
+                    return self.error(
+                        f'Cannot find one or more groups with IDs: {group_ids}.'
+                    )
+
+                if groups:
+                    time_interval.groups = time_interval.groups + groups
+
+            if 'time_interval' in data:
+                setattr(time_interval, 'time_interval', data['time_interval'])
```

Comment:
Why not `time_interval.time_interval = data['time_interval']` ?

Reply:
Good idea.